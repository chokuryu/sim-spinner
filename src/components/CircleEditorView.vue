<template>
  <!--Unique Parameter For Circle-->
  <div class="siimple-grid">
    <div class="siimple-grid-row">
      <div class="siimple-grid-col siimple-grid-col--6 siimple-grid-col-md--12 siimple-grid-col-sm--12">

        <div class="row siimple-label" ref="name_frame">
          <label>Name:</label>
          <div><input type="text" class="siimple-input" v-model="name" value=""/></div>
        </div>
        <div class="row siimple-label" ref="type_frame">
          <label>Type:</label>
          <div><input type="text" class="siimple-input dis" v-model="type" value="" readonly/></div>
        </div>
        <div class="row siimple-label" ref="radius_frame">
          <label>Radius:</label>
          <div><input type="range" max="100" min="10" step="10" class="siimple-input" v-model.number="radius" value=""/>
              <span>{{ radius }}</span><span class="afterfield">px</span></div>
        </div>
        <div class="row siimple-label" ref="speed_frame">
          <label>Speed:</label>
          <div><input type="range" max="2" min="0.1" step="0.1" class="siimple-input" v-model.number="speed" value=""/>
              <span>{{ speed }}</span><span class="afterfield">s</span></div>
        </div>
        <div class="row siimple-label" ref="posY_frame">
          <label>Posision:</label>
          <div><input type="range" max="50" min="-50" step="10" class="siimple-input" v-model.number="posY" value=""/>
              <span>{{ posY }}</span><span class="afterfield"> px</span></div>
        </div>

      </div>
      <div class="siimple-grid-col siimple-grid-col--6 siimple-grid-col-md--12 siimple-grid-col-sm--12">

        <div class="row siimple-label" ref="colorR_frame">
          <label>color R:</label>
          <div><input type="range" max="255" min="0" step="1" class="siimple-input" v-model.number="colorR" value=""/>
              <span>({{ colorR }})</span></div>
        </div>
        <div class="row siimple-label" ref="colorG_frame">
          <label>color G:</label>
          <div><input type="range" max="255" min="0" step="1" class="siimple-input" v-model.number="colorG" value=""/>
              <span>({{ colorG }})</span></div>
        </div>
        <div class="row siimple-label" ref="colorB_frame">
          <label>color B:</label>
          <div><input type="range" max="255" min="0" step="1" class="siimple-input" v-model.number="colorB" value=""/>
              <span>({{ colorB }})</span></div>
        </div>
        <div class="row siimple-label" ref="thin_frame">
          <label>Thin:</label>
          <div><input type="range" max="10" min="1" step="1" class="siimple-input" v-model.number="thin" value=""/>
              <span>({{ thin }})</span><span class="afterfield">px</span></div>
        </div>

      </div>
    </div>
  </div>
</template>

<script>
import Store from '@/scripts/Store'
import FactorDefinition from '@/scripts/FactorDefinition'
const FACTOR_TYPE = 'CIRCLE'

export default {
  name: 'CircleEditorView',
  components: {

  },
  props: ['fid'],
  data () {
    return Store.referFactor(this.fid)
  },

  computed: {

  },
  watch: (()=>{
    let PROPS = FactorDefinition.getEditorObserbedParameter(FACTOR_TYPE)
    let watchMethods = Store.createParameterWatcheMethods(PROPS)
    return watchMethods
  })(),
  methods: {

  }
}

</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped>


</style>
