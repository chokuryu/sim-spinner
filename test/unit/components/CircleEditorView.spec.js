
import { shallowMount } from '@vue/test-utils'
import sinon from 'sinon'

import Store from '@/scripts/Store'
import '../helper/__SetupStore'
import CircleEditorView from '@/components/CircleEditorView'


//console.log(Store.a)


describe('CircleEditorView - unit', () => {

  Store.destroy()
  Store.initializeForTest()

  const mock = {
    id: 'fake',
    name: 'circle1',
    type: 'CIRCLE',
    radius: 60,
    speed: 0.6,
    posY: 0,
    colorR: 75,
    colorG: 164,
    colorB: 230,
    thin: 2
  }
  const wrapper = shallowMount(CircleEditorView, {
    data () {
      return mock
    }
  })
  const wEl = wrapper.find({ ref: 'colorR_frame' })

  it('should render correct contents', () => {
    expect(wrapper.is(CircleEditorView)).toBe(true)
    expect(wEl.find('input').element.value).toBe('75')
    //
    expect(wrapper.element).toMatchSnapshot();
  })

  it('should be reactive', () => {
    mock.colorR += 2
    expect(wEl.find('input').element.value).toBe('77')
  })

})


describe('CircleEditorView - with store', () => {

  Store.destroy()
  Store.initializeForTest()

  const fid = 'f_1'
  const wrapper = shallowMount(CircleEditorView, {
    propsData: {
      fid: fid
    }
  })

  it('should render for the store data', () => {
    expect(wrapper.is(CircleEditorView)).toBe(true)
    expect(typeof wrapper.vm.colorR).toBe('number')
  })

  it('publish update info for store when value changed', () => {
    const spy = sinon.stub()
    Store.subscribeFactorUpdate(fid, () => {
      spy()
    })
    expect(spy.calledOnce).toBe(false)
    wrapper.vm.colorR = 50
    expect(spy.calledOnce).toBe(true)
  })

})
