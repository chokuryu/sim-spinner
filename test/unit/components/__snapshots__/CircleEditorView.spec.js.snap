// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleEditorView - unit should render correct contents 1`] = `
<div
  class="siimple-grid"
>
  <div
    class="siimple-grid-row"
  >
    <div
      class="siimple-grid-col siimple-grid-col--6 siimple-grid-col-md--12 siimple-grid-col-sm--12"
    >
      <div
        class="row siimple-label"
      >
        <label>
          Name:
        </label>
         
        <div>
          <input
            class="siimple-input"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          Type:
        </label>
         
        <div>
          <input
            class="siimple-input dis"
            readonly="readonly"
            type="text"
            value=""
          />
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          Radius:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="100"
            min="10"
            step="10"
            type="range"
            value=""
          />
           
          <span>
            60
          </span>
          <span
            class="afterfield"
          >
            px
          </span>
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          Speed:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="2"
            min="0.1"
            step="0.1"
            type="range"
            value=""
          />
           
          <span>
            0.6
          </span>
          <span
            class="afterfield"
          >
            s
          </span>
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          Posision:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="50"
            min="-50"
            step="10"
            type="range"
            value=""
          />
           
          <span>
            0
          </span>
          <span
            class="afterfield"
          >
             px
          </span>
        </div>
      </div>
    </div>
     
    <div
      class="siimple-grid-col siimple-grid-col--6 siimple-grid-col-md--12 siimple-grid-col-sm--12"
    >
      <div
        class="row siimple-label"
      >
        <label>
          color R:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="255"
            min="0"
            step="1"
            type="range"
            value=""
          />
           
          <span>
            (75)
          </span>
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          color G:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="255"
            min="0"
            step="1"
            type="range"
            value=""
          />
           
          <span>
            (164)
          </span>
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          color B:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="255"
            min="0"
            step="1"
            type="range"
            value=""
          />
           
          <span>
            (230)
          </span>
        </div>
      </div>
       
      <div
        class="row siimple-label"
      >
        <label>
          Thin:
        </label>
         
        <div>
          <input
            class="siimple-input"
            max="10"
            min="1"
            step="1"
            type="range"
            value=""
          />
           
          <span>
            (2)
          </span>
          <span
            class="afterfield"
          >
            px
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
